extensions/git/src/git.ts

```diff
@@ -1997,6 +2004,27 @@ export class Repository {
 		}
 	}
 
+	async getHEADFS(): Promise<Ref | undefined> {
+		try {
+			const raw = await fs.readFile(path.join(this.dotGit.commonPath ?? this.dotGit.path, 'HEAD'), 'utf8');
+
+			// Branch
+			const branchMatch = raw.match(/^ref: refs\/heads\/(?<name>.*)$/m);
+			if (branchMatch?.groups?.name) {
+				return { name: branchMatch.groups.name, commit: undefined, type: RefType.Head };
+			}
+
+			// Detached
+			const commitMatch = raw.match(/^(?<commit>[0-9a-f]{40})$/m);
+			if (commitMatch?.groups?.commit) {
+				return { name: undefined, commit: commitMatch.groups.commit, type: RefType.Head };
+			}
+		}
+		catch (err) { }
```

Comment:
I suggest changing the function type to `getHEADFS(): Promise<Ref>`, removing this `try catch` block, let the exception bubble to `getHEAD` and log it as a warning when it happens, from `getHEAD`.

Reply:
Maybe I am missing something but if we were to do that, if any error occurs in `getHEADFS()` would results in us not attempting to use `git.exe` to get the branch name but would just directly jump into the `catch()` and we would only get the commit.

Reply:
Yes, you would need to refactor `getHEAD` as well. But it would let you have a clean `getHEADFS` function, that actually throws when something goes wrong. That's valuable to have, in case other places of the code invoke it in the future and are interested in that case.

Reply:
Pushed an update addressing the feedback.